### PR TITLE
Add batch Load Api

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?

add batch load API to worker

### Why are the changes needed?
more efficient distributedload

### CODE REVIEW

File: core/transport/src/main/proto/grpc/block_worker.proto [hunk 1]
```diff
@@ -131,6 +132,43 @@ message CacheRequest {
...
+}
+
+enum TaskStatus {
+  SUCCESS = 0;
+  FAILURE = 1;
+  PARTIAL_FAILURE = 2;
+}
+
+message BlockStatus {
+  // The status code, which should be an enum value of [google.rpc.Code][google.rpc.Code].
```
Comment (luzhang6) [inline core/transport/src/main/proto/grpc/block_worker.proto:165]: Do we need to add a block id here to indicate status for that block?

File: core/transport/src/main/proto/grpc/block_worker.proto [hunk 2]
```diff
@@ -131,6 +132,43 @@ message CacheRequest {
...
 
+// Request for load a block into alluxio
+// next available id: 4
+message LoadRequest {
+  repeated File files = 1;
+  optional string user = 2;
+  optional int64 bandwidth = 3;
+}
+
+message File{
```
Comment (luzhang6) [inline core/transport/src/main/proto/grpc/block_worker.proto:143]: maybe rename File to FileBlocks?

**Comment by luzhang6** [comment]
Left two comments, LGTM overall!

File: core/transport/src/main/proto/grpc/block_worker.proto (hunk 1 above)
Comment (jja725) [reply core/transport/src/main/proto/grpc/block_worker.proto:165]: I think it would go with each block so no need to put id here causing duplication

Comment (luzhang6) [reply core/transport/src/main/proto/grpc/block_worker.proto:165]: i see, thanks

**Review by luzhang6 — APPROVED** [review]

File: core/transport/src/main/proto/grpc/block_worker.proto [hunk 3]
```diff
@@ -131,6 +132,43 @@ message CacheRequest {
...
+}
+
+message FileBlocks{
+  optional string ufs_path = 1;
+  repeated int64 block_id = 2;
+  required int64 block_size = 3;
+}
+
+message LoadResponse {
+  repeated FileStatus file_status = 1;
```
Comment (rongrong) [inline core/transport/src/main/proto/grpc/block_worker.proto:150]: switch the order of the two.

File: core/transport/src/main/proto/grpc/block_worker.proto [hunk 4]
```diff
@@ -131,6 +132,43 @@ message CacheRequest {
...
+  FAILURE = 1;
+  PARTIAL_FAILURE = 2;
+}
+
+message BlockStatus {
+  // The status code, which should be an enum value of [google.rpc.Code][google.rpc.Code].
+  required int32 code = 1;
+
+  // A developer-facing error message
+  optional string message = 2;
```
Comment (rongrong) [inline core/transport/src/main/proto/grpc/block_worker.proto:169]: Let's see whether we can directly use Status, if not, we need to define a new proto for this. Only a string won't be enough.

File: core/transport/src/main/proto/grpc/block_worker.proto [hunk 5]
```diff
@@ -131,6 +132,43 @@ message CacheRequest {
...
+enum TaskStatus {
+  SUCCESS = 0;
+  FAILURE = 1;
+  PARTIAL_FAILURE = 2;
+}
+
+message BlockStatus {
+  // The status code, which should be an enum value of [google.rpc.Code][google.rpc.Code].
+  required int32 code = 1;
+
```
Comment (rongrong) [inline core/transport/src/main/proto/grpc/block_worker.proto:167]: Consider adding a `retryable` boolean.

File: core/common/src/main/java/alluxio/worker/block/BlockWorker.java [hunk 6]
```diff
@@ -210,6 +212,13 @@ void requestSpace(long sessionId, long blockId, long additionalBytes)
    */
   void cache(CacheRequest request) throws AlluxioException, IOException;
 
+  /**
+   * load blocks into alluxio.
+   * @param request load request
+   * @return load status
+   */
+  List<List<BlockStatus>> load(LoadRequest request);
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/worker/block/BlockWorker.java:220]: This should be change to return List<BlockStatus> as well? or just LoadResponse?

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerClientServiceHandler.java [hunk 7]
```diff
@@ -166,6 +170,15 @@ public void cache(CacheRequest request, StreamObserver<CacheResponse> responseOb
     }, "cache", "request=%s", responseObserver, request);
   }
 
+  @Override
+  public void load(LoadRequest request, StreamObserver<LoadResponse> responseObserver) {
+    RpcUtils.call(LOG, () -> {
+      LoadResponse.Builder response = LoadResponse.newBuilder();
+      List<List<BlockStatus>> summary = mBlockWorker.load(request);
```
Comment (rongrong) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerClientServiceHandler.java:177]: summary is not used.

File: core/transport/src/main/proto/grpc/block_worker.proto [hunk 8]
```diff
@@ -131,6 +132,41 @@ message CacheRequest {
...
+  PARTIAL_FAILURE = 2;
+}
+
+message BlockStatus {
+  required int64 block_id = 1;
+  // The status code, which should be an enum value of [google.rpc.Code][google.rpc.Code].
+  required int32 code = 2;
+  // A developer-facing error message
+  optional string message = 3;
+  optional bool retryable = 4;
```
Comment (rongrong) [inline core/transport/src/main/proto/grpc/block_worker.proto:167]: let's make this required as well. Otherwise it's unclear what does missing mean.

**Review by rongrong — APPROVED** [review]

File: core/transport/src/main/proto/grpc/block_worker.proto (hunk 8 above)
Comment (jja725) [reply core/transport/src/main/proto/grpc/block_worker.proto:167]: I'm a little concerned about back compatibility that we might want to change this field to something like retry info in the future. And default value is false so I would imagine we only specify this field when it's retryable in the very specific case

**Comment by rongrong** [comment]
alluxio-bot, merge this please

**Pull request merged**